docs: guide -> ngmodule: fix markdown fomatting

tiangolo opened the pull request:
Fix little markdown formatting errors.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ :heavy_check_mark:  ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ N/A ] Tests for the changes have been added (for bug fixes / features)
- [ :heavy_check_mark:  ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

At some places in the guide there are little formatting errors, like in: https://angular.io/guide/ngmodule#provide-the-_contactservice_

It appears like `Provide the _ContactService_`.

Issue Number: N/A


## What is the new behavior?

The content is formatted to be displayed correctly when there are literal HTML elements.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

tiangolo committed:
docs: guide -> ngmodule: fix markdown fomatting

Fix little markdown formatting errors.

petebacondarwin commented on aio/content/guide/ngmodule.md:
```diff
@@ -383,7 +383,7 @@ You must import those modules before you can use their directives.
 a form component that imports form support from the Angular `FormsModule`.
 
-<h3 class="no-toc">Add the _ContactComponent_</h3>
+<h3 class="no-toc">Add the <em>ContactComponent</em></h3>
```
I would argue that actually what we want is to use `<code>ContactComponent</code>` here.

tiangolo replied:
@petebacondarwin That's even more reasonable. 

I updated this and all the other headers / titles that would use the equivalent of an `<em>` to refer to a `<code>` element. Also in markdown syntax. So, I changed all the 

```
_CodeElement_
```

headers to 

```
`CodeElement`
```

I didn't update those that do not refer to a code element, as the `Bootstrapping in _main.ts_` header.

tiangolo committed:
docs: update NgModules guide docs, code style

Change titles in Guide -> NgModules that refer to code elements to use <code> (or "`") syntax instead of <em> (or "_") syntax.

mary-poppins commented:
You can preview abdc401 at https://pr18405-abdc401.ngbuilds.io/.
You can preview ff57514 at https://pr18405-ff57514.ngbuilds.io/.

tiangolo commented:
This PR was created more than 6 months ago. I don't know if someone will check it, so I guess it's maybe not worth checking and fixing it. It might be that the changes don't apply anymore.

So, unless I hear any feedback or interest on this PR from anyone here, I will just leave it here waiting.

gkalpak commented:
Ooops, sorry for this, @tiangolo. We should have gotten to this sooner 😒 
We appreciate all contributions (:heart:) and try to stay on top of the PR queue, but sometimes things may slip through the cracks 😞

##
The `ngmodule` guide has been split up into several separate guides (under [Fundamentals\NgModules][1]). I checked a couple of this PR's changes and they have indeed been addressed (or become obsolete) - but I didn't check all of them.

There still two PRs in progress with some fixes/changes to these guides (#21517, #21589), but if you spot anything else (or if some of the changes still apply), please submit a new PR. We'll make sure it gets handled much more promptly this time :smiley: 

[1]: https://angular.io/guide/ngmodules

tiangolo commented:
Thank you very much for the warm feedback!
